jpa/persistence-cdi-extension/src/main/java/org/jboss/as/jpa/beanmanager/IntegratePersistenceAfterBeanDiscovery.java

```diff
@@ -0,0 +1,444 @@
+                    return integrationWithCDIBag.getEntityManagerFactory();
+                }
+        );
+    }
+
+    private void criteriaBuilder(
+            AfterBeanDiscovery afterBeanDiscovery,
+            PersistenceUnitMetadata persistenceUnitMetadata,
+            List<String> qualifiers,
+            IntegrationWithCDIBag integrationWithCDIBag) {
+
+        String scope = dependentScoped;
+
+        BeanConfigurator<CriteriaBuilder> beanConfigurator = afterBeanDiscovery.addBean();
+        beanConfigurator.addTransitiveTypeClosure(CriteriaBuilder.class);
+
+        Class<? extends Annotation> scopeAnnotation = null;
+        try {
+            scopeAnnotation = persistenceUnitMetadata.getClassLoader().loadClass(scope).asSubclass(Annotation.class);
+        } catch (ClassNotFoundException e) {
+            throw JpaLogger.ROOT_LOGGER.classNotFound(e, "CriteriaBuilder", persistenceUnitMetadata.getPersistenceUnitName(), persistenceUnitMetadata.getScopedPersistenceUnitName());
+        }
+        beanConfigurator.scope(scopeAnnotation);
+
+        for (String qualifier : qualifiers) {
+            final Class<? extends Annotation> qualifierType;
+            try {
+                qualifierType = persistenceUnitMetadata.getClassLoader()
+                        .loadClass(qualifier)
+                        .asSubclass(Annotation.class);
+            } catch (ClassNotFoundException e) {
+                throw JpaLogger.ROOT_LOGGER.classNotFound(e, "CriteriaBuilder", persistenceUnitMetadata.getPersistenceUnitName(), persistenceUnitMetadata.getScopedPersistenceUnitName());
+            }
+            // beanConfigurator.addQualifier(qualifierType);
+            beanConfigurator.addQualifier(ScopeProxy.createProxy(qualifierType));
+        }
+
+        beanConfigurator.beanClass(CriteriaBuilder.class);
+        beanConfigurator.produceWith(c -> {
+                    return integrationWithCDIBag.getEntityManagerFactory().getCriteriaBuilder();
+                }
+        );
+
+    }
+
+    private void persistenceUnitUtil(
+            AfterBeanDiscovery afterBeanDiscovery,
+            PersistenceUnitMetadata persistenceUnitMetadata,
+            List<String> qualifiers,
+            IntegrationWithCDIBag integrationWithCDIBag) {
+
+        String scope = dependentScoped;
+
+        BeanConfigurator<PersistenceUnitUtil> beanConfigurator = afterBeanDiscovery.addBean();
+        beanConfigurator.addTransitiveTypeClosure(PersistenceUnitUtil.class);
+
+
+        Class<? extends Annotation> scopeAnnotation = null;
+        try {
+            scopeAnnotation = persistenceUnitMetadata.getClassLoader().loadClass(scope).asSubclass(Annotation.class);
+        } catch (ClassNotFoundException e) {
+            throw JpaLogger.ROOT_LOGGER.classNotFound(e, "PersistenceUnitUtil", persistenceUnitMetadata.getPersistenceUnitName(), persistenceUnitMetadata.getScopedPersistenceUnitName());
+        }
+        beanConfigurator.scope(scopeAnnotation);
```

Comment:
This could simply be `beanConfiguator.scope(Dependent.class)`. This would save the class loading as the scope from this producer cannot be changed.